### PR TITLE
sends contact notifications after user sets their Name 

### PR DESCRIPTION
**Pull request by joshm1**
also factored some User business logic out of the HomeController

### CODE REVIEW

**Comment by joshm1** [comment]
@stkem

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/HomeController.scala [hunk 1]
```diff
@@ -248,18 +248,10 @@ class HomeController @Inject() (
...
-        senderUserId <- {
-          invitationRepo.save(invite.withState(InvitationStates.JOINED))
-          invite.senderUserId
-        }
-      } yield senderUserId
-    }
     SafeFuture {
-      userCommander.tellAllFriendsAboutNewUser(request.user.id.get, toBeNotified.toSet.toSeq)
+      userCommander.tellAllFriendsAboutNewUser(request.user.id.get)
+      userCommander.tellUsersWithContactOfNewUserImmediate(request.user)
```
Comment (stkem) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/HomeController.scala:253]: I see no matching removed line. Is this now being called in two places or was it not called at all before?

**Comment by stkem** [comment]
made one comment, lgtm otherwise

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/HomeController.scala (hunk 1 above)
Comment (joshm1) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/HomeController.scala:253]: doh. good catch!

**Pull request merged**